avocado/utils/linux_modules.py

```diff
@@ -92,11 +99,31 @@ def loaded_module_info(module_name):
 
     :param module_name: Name of module to search for
     :type module_name: str
-    :return: Dictionary of module info, name, size, submodules if present
+    :return: Dictionary of module name, size, submodules if present, filename,
+             version, number of modules using it, list of modules it is
+             dependent on, list of params
     :rtype: dict
     """
     l_raw = process.system_output('/sbin/lsmod')
-    return parse_lsmod_for_module(l_raw, module_name)
+    modinfo_dic = parse_lsmod_for_module(l_raw, module_name)
+    output = process.system_output("/sbin/modinfo %s" % module_name)
+    if output:
+        param_list = []
+        for line in output.splitlines():
+            items = line.split()
+            key = items[0].rstrip(':')
```

Comment:
I would try to protect the parsing code against malformed command output. For example, what if modinfo spits some error message instead of the expected output? We should catch things like IndexError, since we are refering to a list that might not have all the elements you expect.